CMake/AbseilDll.cmake

```diff
@@ -522,6 +522,25 @@ function(absl_make_dll)
     PRIVATE
       ${ABSL_DEFAULT_LINKOPTS}
   )
+
+  if(ABSL_PROPAGATE_CXX_STD)
+    # Abseil libraries require C++11 as the current minimum standard.
+    # Top-level application CMake projects should ensure a consistent C++
+    # standard for all compiled sources by setting CMAKE_CXX_STANDARD.
+    target_compile_features(abseil_dll PUBLIC cxx_std_${ABSL_CXX_STANDARD})
```

Comment:
Is `ABSL_CXX_STANDARD` always defined?  I think the default is set here:

https://github.com/abseil/abseil-cpp/blob/4a1ccf16ed98c876bf1e1985afb080baeff5101f/absl/copts/AbseilConfigureCopts.cmake#L97

And that looks like it can be the empty string?  And if is the empty string, then the C++ standard in effect depends on the compiler (and compiler *version*).

Reply:
@coryan see the example patterns in the Abseil CMake README here:
https://github.com/abseil/abseil-cpp/tree/master/CMake#step-by-step-instructions

I think library authors (including Abseil itself) should NOT set `CMAKE_CXX_STANDARD`, which should be left up to the end application project.

When a library project is being directly developed or CI tested, then it can safely set those things without affecting its clients (which is the pattern the `if(CMAKE_SOURCE_DIR STREQUAL my_lib_project_SOURCE_DIR)` check in the 2nd README example shows; it can also be set via the `cmake` command-line in build scripts).